### PR TITLE
Update the playground documentation

### PR DESCRIPTION
**Pull request by JeffGuKang**
I completed the live docs using playground according to the latest version. 

The PR should summarize what was changed and why. Here are some questions to
help you if you're not sure:

 - What behavior was changed?
Update the playground documentation
 - What code was refactored / updated to support this change?
Add sections of usage from README
 - What issues are related to this PR? Or why was this change introduced?

Checklist - While not every PR needs it, new features should consider this list:

 - [v] Does this have tests?
 - [v] Does this have documentation?
 - [] Does this break the public API (Requires major version bump)? no problem
 - [] Is this a new feature (Requires minor version bump)? no problem

### CODE REVIEW

File: Example/Playground.playground/Contents.swift [hunk 1]
```diff
@@ -1,98 +1,416 @@
...
+	//Do something you want
+	print("\(index): \(subJson)")
+}
+
+/*:
+### Error
+
+SwiftyJSON 4.x
+
+SwiftyJSON 4.x introduces an enum type called `SwiftyJSONError`, which includes `unsupportedType`, `indexOutOfBounds`, `elementTooDeep`, `wrongType`, `notExist` and `invalidJSON`, at the same time, `ErrorDomain` are being replaced by `SwiftyJSONError.errorDomain`. Note: Those old error types are deprecated in SwiftyJSON 4.x and will be removed in the future release.
```
[the remark this answers is not in the document]
Comment (JeffGuKang) [reply Example/Playground.playground/Contents.swift]: Can be ignored

File: Example/Playground.playground/Contents.swift [hunk 2]
```diff
@@ -1,98 +1,416 @@
...
+let user: JSON = ["username" : "Steve", "password": "supersecurepassword"]
+let auth: JSON = [
+	"user": user.object, //use user.object instead of just user
+	"apikey": "supersecretapitoken"
+]
+
+/*:
+### Merging
+
+It is possible to merge one JSON into another JSON. Merging a JSON into another JSON adds all non existing values to the original JSON which are only present in the other JSON.
```
[the remark this answers is not in the document]
Comment (JeffGuKang) [reply Example/Playground.playground/Contents.swift]: Can be ignored

File: Example/Playground.playground/Contents.swift [hunk 3]
```diff
@@ -1,98 +1,416 @@
...
+
+If both JSONs contain a value for the same key, mostly this value gets overwritten in the original JSON, but there are two cases where it provides some special treatment:
```
[the remark this answers is not in the document]
Comment (JeffGuKang) [reply Example/Playground.playground/Contents.swift]: Can be ignored

File: Example/Playground.playground/Contents.swift [hunk 4]
```diff
@@ -1,98 +1,416 @@
...
+
+- In case of both values being a JSON.Type.array the values form the array found in the other JSON getting appended to the original JSON's array value.
```
[the remark this answers is not in the document]
Comment (JeffGuKang) [reply Example/Playground.playground/Contents.swift]: Can be ignored

File: Example/Playground.playground/Contents.swift [hunk 5]
```diff
@@ -1,98 +1,416 @@
...
+- In case of both values being a JSON.Type.dictionary both JSON-values are getting merged the same way the encapsulating JSON is merged.
+
+In case, where two fields in a JSON have a different types, the value will get always overwritten.
+
+There are two different fashions for merging: merge modifies the original JSON, whereas merged works non-destructively on a copy.
```
[the remark this answers is not in the document]
Comment (JeffGuKang) [reply Example/Playground.playground/Contents.swift]: Can be ignored

File: Example/Playground.playground/Contents.swift [hunk 6]
```diff
@@ -1,98 +1,416 @@
...
+
+/*:
+### Merging
+
+It is possible to merge one JSON into another JSON. Merging a JSON into another JSON adds all non existing values to the original JSON which are only present in the other JSON.
+
+If both JSONs contain a value for the same key, mostly this value gets overwritten in the original JSON, but there are two cases where it provides some special treatment:
+
+- In case of both values being a JSON.Type.array the values form the array found in the other JSON getting appended to the original JSON's array value.
+- In case of both values being a JSON.Type.dictionary both JSON-values are getting merged the same way the encapsulating JSON is merged.
```
[the remark this answers is not in the document]
Comment (JeffGuKang) [reply Example/Playground.playground/Contents.swift]: Can be ignored

**Comment by JeffGuKang** [comment]
Violations are fixed

**Comment by wongzigii** [comment]
Thanks @JeffGuKang

**Pull request merged**